packages/liveblocks-react-ui/src/primitives/AiMessage/index.tsx

```diff
@@ -0,0 +1,169 @@
+import type {
+  AiToolDefinitionRenderProps,
+  AiToolInvocationPart,
+  Json,
+  MessageId,
+} from "@liveblocks/core";
+import { kInternal } from "@liveblocks/core";
+import { useClient } from "@liveblocks/react";
+import { useSignal } from "@liveblocks/react/_private";
+import { Slot } from "@radix-ui/react-slot";
+import {
+  createContext,
+  forwardRef,
+  useCallback,
+  useContext,
+  useMemo,
+} from "react";
+
+import type {
+  AiMessageContentComponents,
+  AiMessageContentProps,
+} from "./types";
+
+const AI_MESSAGE_CONTENT_NAME = "AiMessageContent";
+
+const defaultMessageContentComponents: AiMessageContentComponents = {
+  TextPart: () => "Default TextPart is not implemented yet",
+  ReasoningPart: () => "Default ReasoningPart is not implemented yet",
+  UploadedImagePart: () => "Default UploadedImagePart is not implemented yet",
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolInvocationPart
+ * -----------------------------------------------------------------------------------------------*/
+// XXX Rename to AiToolInvocationContext?
+const AiToolDefinitionRenderContext =
+  createContext<AiToolDefinitionRenderProps | null>(null);
+
+// XXX Rename to useAiToolInvocationContext?
+export function useAiToolDefinitionRenderContext() {
+  const context = useContext(AiToolDefinitionRenderContext);
```

Comment:
I haven't renamed these yet, wanted to get your opinion first, @marcbouchenoire.

Reply:
I think I like `AiToolInvocationContext`/`useAiToolInvocationContext` but maybe `AiToolDefinitionRenderProps` should use the same terms then, `AiToolInvocationRenderProps` (or even `AiToolInvocationProps` since `execute` doesn't get "props" but just "args")?

Reply:
Good point, and I agree 👍